lightbug_http/sys/net.mojo

```diff
@@ -202,3 +242,151 @@ struct SysNet(Net):
 
     fn listen(inout self, network: String, addr: String) raises -> SysListener:
         return self.__lc.listen(network, addr)
+
+
+@value
+@register_passable("trivial")
+struct addrinfo_macos(AnAddrInfo):
+    """
+    For MacOS, I had to swap the order of ai_canonname and ai_addr.
+    https://stackoverflow.com/questions/53575101/calling-getaddrinfo-directly-from-python-ai-addr-is-null-pointer.
+    """
+
+    var ai_flags: c_int
+    var ai_family: c_int
+    var ai_socktype: c_int
+    var ai_protocol: c_int
+    var ai_addrlen: socklen_t
+    var ai_canonname: Pointer[c_char]
+    var ai_addr: Pointer[sockaddr]
+    var ai_next: Pointer[c_void]
+
+    fn __init__() -> Self:
+        return Self(
+            0, 0, 0, 0, 0, Pointer[c_char](), Pointer[sockaddr](), Pointer[c_void]()
+        )
+
+    fn get_ip_address(self, host: String) raises -> in_addr:
+        var host_ptr = to_char_ptr(host)
+        var servinfo = Pointer[Self]().alloc(1)
+        servinfo.store(Self())
+
+        var hints = Self()
+        hints.ai_family = AF_INET
+        hints.ai_socktype = SOCK_STREAM
+        hints.ai_flags = AI_PASSIVE
+
+        var error = getaddrinfo[Self](
+            host_ptr,
+            Pointer[UInt8](),
+            Pointer.address_of(hints),
+            Pointer.address_of(servinfo),
+        )
+        if error != 0:
+            print("getaddrinfo failed")
+            raise Error("Failed to get IP address. getaddrinfo failed.")
+
+        var addrinfo = servinfo.load()
+
+        var ai_addr = addrinfo.ai_addr
+        if not ai_addr:
+            print("ai_addr is null")
+            raise Error(
+                "Failed to get IP address. getaddrinfo was called successfully, but"
+                " ai_addr is null."
+            )
+
+        var addr_in = ai_addr.bitcast[sockaddr_in]().load()
+
+        return addr_in.sin_addr
```

Comment:
This method is the same in both unix & macos structs. Figure when they add the ability to set default function in a trait this can be moved the trait as it's identical for both.